pytorch_lightning/trainer/data_loading.py

```diff
@@ -187,7 +187,7 @@ def _reset_eval_dataloader(self, model: LightningModule,
             dataloaders = [dataloaders]
 
         # add samplers
-        dataloaders = [self.auto_add_sampler(dl, train=False) for dl in dataloaders if dl]
+        dataloaders = [self.auto_add_sampler(dl, train=False) for dl in dataloaders]
```

Comment:
This is a good change but i'm concerned that it'll break something else. Can you change it to if dl != None to be slightly more safe.

Reply:
makes sense, I will make the change

Reply:
for nor rather use `is not None` as it is rather class then a value

Reply:
If the goal is to check for a None value, then `is not None` or `is None` is definitely the correct way. Otherwise, with the equality check, it will invoke `__eq__` on the object which it may or may not have or is overriden.